database/redis/metric.go

```diff
@@ -136,40 +137,92 @@ func (connector *DbConnector) SaveMetrics(metrics map[string]*moira.MatchedMetri
 			}
 
 			var metricEventsChannel = metricEventsChannels[rand.Intn(len(metricEventsChannels))]
-			if err = c.Publish(ctx, metricEventsChannel, event).Err(); err != nil {
-				return err
-			}
+			pipe.SAdd(ctx, metricEventsChannel, event)
 		}
 	}
+
+	if _, err = pipe.Exec(ctx); err != nil {
+		connector.logger.Errorf("Sending metric event error. Error: %v;", err)
+		return err
+	}
+
 	return nil
 }
 
+const (
+	metricEventPopBatchSize   = 100
+	receiveEmptySleepDuration = time.Second
+)
+
 // SubscribeMetricEvents creates subscription for new metrics and return channel for this events
 func (connector *DbConnector) SubscribeMetricEvents(tomb *tomb.Tomb) (<-chan *moira.MetricEvent, error) {
-	metricsChannel := make(chan *moira.MetricEvent, pubSubWorkerChannelSize)
-	dataChannel, err := connector.manageSubscriptions(tomb, metricEventsChannels)
-	if err != nil {
+	responesChannel := make(chan string, pubSubWorkerChannelSize)
+	metricChannel := make(chan *moira.MetricEvent, pubSubWorkerChannelSize)
+
+	ctx := connector.context
+	c := *connector.client
+
+	if err := c.Ping(ctx).Err(); err != nil {
 		return nil, err
 	}
 
+	go func() {
+		<-tomb.Dying()
+		close(responesChannel)
+	}()
+
 	go func() {
 		for {
-			data, ok := <-dataChannel
+			response, ok := <-responesChannel
 			if !ok {
-				connector.logger.Info("No more subscriptions, channel is closed. Stop process data...")
-				close(metricsChannel)
+				close(metricChannel)
 				return
 			}
+
 			metricEvent := &moira.MetricEvent{}
-			if err := json.Unmarshal(data, metricEvent); err != nil {
-				connector.logger.Errorf("Failed to parse MetricEvent: %s, error : %v", string(data), err)
+			if err := json.Unmarshal([]byte(response), metricEvent); err != nil {
+				connector.logger.Errorf("Failed to parse MetricEvent: %s, error : %v", response, err)
 				continue
 			}
-			metricsChannel <- metricEvent
+			metricChannel <- metricEvent
 		}
 	}()
 
-	return metricsChannel, nil
+	for channelIdx := 0; channelIdx < len(metricEventsChannels); channelIdx++ {
+		metricEventsChannel := metricEventsChannels[channelIdx]
```

Comment:
but this is not channel as channel... 
i saw that this was naming before urs changes, but hm...

not necessary, i think

Reply:
Redis documentation name it channel, so i think it is correct